File: hanoi.ex
defmodule Hanoi do
  def hanoi(n,x,y,z) do
    if n>0, do: hanoi(n-1,x,z,y)
    IO.puts "#{x}=>#{y}"
    if n>0, do: hanoi(n-1,x,z,y)
  end
end

Hanoi.hanoi(String.to_integer(IO.gets ""),"a","b","c")

File: hanoi.rb
def hanoi(n,x,y,z)
  hanoi(n-1,x,z,y) if n>1
  print("#{x}=>#{y}\n")
  hanoi(n-1,z,y,x) if n>1
end

hanoi(gets.chomp.to_i,"a","b","c")
